dicodile/update_z/tests/test_distributed_sparse_encoder.py

```diff
@@ -25,29 +26,38 @@ def test_distributed_sparse_encoder():
                   freeze_support=False, warm_start=False, random_state=27)
 
     X = rng.randn(n_channels, n_times)
-    D = rng.randn(n_atoms, n_channels, n_times_atom)
-    sum_axis = tuple(range(1, D.ndim))
-    D /= np.sqrt(np.sum(D * D, axis=sum_axis, keepdims=True))
-    DtD = compute_DtD(D)
+    if not rank1:
+        D = rng.randn(n_atoms, n_channels, *atom_support)
+        sum_axis = tuple(range(1, D.ndim))
+        D /= np.sqrt(np.sum(D * D, axis=sum_axis, keepdims=True))
+    else:
+        u = rng.randn(n_atoms, n_channels)
+        v = rng.randn(n_atoms, *atom_support)
+        # XXX normalize?
+        D = u, v
 
+    DtD = compute_DtD(D)
     encoder = DistributedSparseEncoder(n_workers=2)
-
     encoder.init_workers(X, D, reg, params, DtD=DtD)
 
-    with pytest.raises(ValueError, match=r"pre-computed value DtD"):
-        encoder.set_worker_D(D)
+    # XXX should that moved to a separate test?
```

Comment:
btw @tomMoral do you have an opinion on this, move to a separate test or keep as-is?